server/src/postgis/best_path.rs

```diff
@@ -0,0 +1,392 @@
+        };
+
+        if time_end < time_start {
+            return Err(PathError::InvalidTimeWindow);
+        }
+
+        if time_end < Utc::now() {
+            return Err(PathError::InvalidEndTime);
+        }
+
+        Ok(PathRequest {
+            node_start_id,
+            node_uuid_end,
+            time_start,
+            time_end,
+        })
+    }
+}
+
+/// Get the best path from a vertiport to another vertiport
+async fn best_path_vertiport_source(
+    stmt: tokio_postgres::Statement,
+    client: deadpool_postgres::Client,
+    request: PathRequest,
+) -> Result<Vec<tokio_postgres::Row>, PathError> {
+    let Ok(node_start_id) = Uuid::parse_str(&request.node_start_id) else {
+        postgis_error!("(best_path_vertiport_source) could not parse start node id into UUID: {}", request.node_start_id);
+        return Err(PathError::InvalidStartNode);
+    };
+
+    match client
+        .query(
+            &stmt,
+            &[
+                &node_start_id,
+                &request.node_uuid_end,
+                &request.time_start,
+                &request.time_end,
+            ],
+        )
+        .await
+    {
+        Ok(results) => Ok(results),
+        Err(e) => {
+            println!(
+                "(best_path_vertiport_source) could not request routes: {}",
+                e
+            );
+            Err(PathError::Unknown)
+        }
+    }
+}
+
+/// Get the best path from a aircraft to another vertiport
+async fn best_path_aircraft_source(
+    stmt: tokio_postgres::Statement,
+    client: deadpool_postgres::Client,
+    request: PathRequest,
+) -> Result<Vec<tokio_postgres::Row>, PathError> {
+    match client
+        .query(
+            &stmt,
+            &[
+                &request.node_start_id,
+                &request.node_uuid_end,
+                &request.time_start,
+                &request.time_end,
+            ],
+        )
+        .await
+    {
+        Ok(results) => Ok(results),
+        Err(e) => {
+            println!(
+                "(best_path_aircraft_source) could not request routes: {}",
+                e
+            );
+
+            Err(PathError::Unknown)
+        }
+    }
+}
+
+/// The purpose of this initial search is to verify that a flight between two
+///  vertiports is physically possible.
+///
+/// A flight is physically impossible if the two vertiports cannot be
+///  connected by a series of lines such that the aircraft never runs out
+///  of charge.
+///
+/// No-Fly zones can extend flights, isolate aircraft, or disable vertiports entirely.
+#[cfg(not(tarpaulin_include))]
+pub async fn best_path(
+    path_type: PathType,
+    request: BestPathRequest,
+    pool: deadpool_postgres::Pool,
+) -> Result<Vec<PathSegment>, PathError> {
+    let request = PathRequest::try_from(request)?;
+    let client = match pool.get().await {
+        Ok(client) => client,
+        Err(e) => {
+            println!("(best_path) could not get client from pool.");
```

Comment:
Should be: `postgis_error!(`

Reply:
https://github.com/Arrow-air/svc-gis/commit/2b93948539015c3003bde9fba7385a81e9e6f4dd